docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+
+` mlnet new @my_cli_config_args.rsp`
+
+
+(*Release 0.2 examples*)
+
+Simplest command where the tool infers the type of ML taks to perform based on the data:
+
+` mlnet new --dataset "/MyDataSets/Cars.csv"`
+
+Create and train a model based on parameters specified in the .rsp file plus more advanced model settings in the .yaml file:
+
+` mlnet new @my_cli_config_args.rsp --model-settings-file "./my_model_settings.yaml"  `
+
+
+(*Release v1.0 examples*)
+
+CLI command specifying to run AutoML compute in Azure's cloud 
+
+` mlnet new --ml-task BinaryClassification --dataset "/MyDataSets/Cars.csv" --label-column-name "InsuranceRisk" --azure-automl-service-key "key-value"`
+
+
+
+Syntax should be aligned to [dotnet new](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new?tabs=netcore21#synopsis).
+
+#### Syntax
+
+```console
+mlnet new 
+
+--------------- (v0.1) -------------------
+
+--ml-task <value>
+
+--dataset <value>
+[
+ [--validation-dataset <value>]
+  --test-dataset <value>
+]
+
+--label-column-name <value>
+|
+--label-column-index <value>
+
+[--has-header <value>]
+
+[--max-exploration-time <value>]
+[--verbosity <value>]
+
+[--name <value>]
+[--list-ml-tasks]
+[-h|--help]
+
+-------------- (v0.2) --------------------
+[--ml-task <value>]
+[--model-settings-file <value>]
+[<RSP FILE NAME>]
+[--best-models-count]
+[--app-type <value>]
+[--output-folder <value>]
+[--force]
+
+-------------- (v1.0) --------------------
+[--list-app-types]
+[--debug-log-file <value>]
+------------------------------------------
+
+-------------- (vNext) -------------------
+[-lang|--language <value>]
+------------------------------------------
+
+```
+
+#### Description
+
+The `mlnet new` command provides a CLI oriented way to create projects or solutions such as:
+
+- Create a single project (console app) with:
+    - *Training ML.NET code:* One seggregated method per ranked model, but part of the same console app.
+    - *Scoring/consuming ML.NET code:* One seggregated method per ranked model, but part of the same console app.
+    - *Trained model files:* Multiple ranked trained models in the form of several .ZIP files.
+
+Optionally (*but in vNext CLI versions such as v1.0 or later*), depending on the used `--app-type argument`, it could also create a solution, such as:
+
+- Create a solution (multiple projects) with:
+    - *Training project:* Console project with model-training ML.NET code
+    - *Common-code project:* Class library project with common code (Data/Observation class, Prediction class, etc.
+    - *End-user-app project:* End-user application type (depending on template) with ML.NET code scoring the model/s.
+    - *Trained models:* Multiple ranked trained models in the form of several .ZIP files.
+
+
+
+### Arguments
+
+Invalid input of arguments should cause it to emit a list of valid inputs and an error message explaining which arg is missing, if that is the case.
```

Comment:
other handling questions:
What happens when we can not infer label? 
How do we tell what we decided the label and features be?